src/DeepCopy/Reflection/ReflectionHelper.php

```diff
@@ -27,8 +27,8 @@ public static function getProperties(\ReflectionClass $ref)
 
         if ($parentClass = $ref->getParentClass()) {
             $parentPropsArr = self::getProperties($parentClass);
-            foreach ($parentPropsArr as $property) {
-                $propsArr[] = $property;
```

Comment:
another fix to still benefit from the performance gain is to simply add the key:

```php
foreach($parentPropsArr as $key => $property) {
  $propsArr[$key] = $property;
}
```